### PR TITLE
Ui tests. Initial commit

### PR DESCRIPTION
**Pull request by E-Kel**

### CODE REVIEW

File: src/test/java/ui/e2e/CheckoutTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import static com.codeborne.selenide.Selenide.open;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static ui.constants.CheckoutConstants.TOTAL_PRICE;
+import static ui.constants.CredentialsValues.EMAIL;
+import static ui.constants.CredentialsValues.PASSWORD;
+
+public class CheckoutTest {
+    public MainPage openMainPage() {
+        return open(URL.PRODUCT_URL, MainPage.class);
```
Comment (yuliiakholina) [inline src/test/java/ui/e2e/CheckoutTest.java:18]: does this method should be really in this class?

File: src/test/java/ui/filters/FiltersTest.java [hunk 2]
```diff
@@ -1,46 +1,39 @@
...
+//    void verifyStyleFilterForDresses() {
+//        assertTrue(
+//                CollectionUtils.isEqualCollection(
+//                        searchPage.clickOnDresses()
+//                                .setStyleFilterToGirly()
+//                                .getElementsAfterFilter(searchPage.getSearchResultName()),
+//                        SearchConstants.RESULT_VALUES_STYLE_SEARCH
+//                )
+//        );
+//    }
```
Comment (4etverochka) [inline src/test/java/ui/filters/FiltersTest.java:38]: What's wrong with this part? Why is it in comment?

File: src/test/java/ui/filters/FiltersTest.java [hunk 3]
```diff
@@ -1,46 +1,39 @@
...
-    void verifyStyleFilterForDresses() {
-        assertTrue(
-                CollectionUtils.isEqualCollection(
-                        searchPage.clickOnDresses()
-                                .setStyleFilterToGirly()
-                                .getElementsAfterFilter(searchPage.getSearchResultName()),
-                        SearchConstants.RESULT_VALUES_STYLE_SEARCH
-                )
-        );
-    }
```
Comment (4etverochka) [inline src/test/java/ui/filters/FiltersTest.java:45]: Why all of this is deleted?

File: src/test/java/ui/e2e/CheckoutTest.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            addressPage.clickProceedToCheckout();
+            shippingPage.clickProceedToCheckout();
+        }
+    }
+
+    private void openSiteAndAddProductToCart() {
+        openMainPage().addProductToCart();
+        productPage.clickAddToCartButton();
+        productPage.closePopupIcon();
+    }
```
Comment (4etverochka) [inline src/test/java/ui/e2e/CheckoutTest.java:73]: Move methods not "Test" methods to another class

File: src/main/java/ui/constants/CheckoutConstants.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package ui.constants;
+
+public class CheckoutConstants {
+    public static String TOTAL_PRICE = "";
+
+    public static void setTotalPrice(String totalPrice) {
```
Comment (4etverochka) [inline src/main/java/ui/constants/CheckoutConstants.java:6]: Using setter for public field is not suitable

**Comment by 4etverochka** [comment]
Use Selenide waiters for each action on elements

File: src/test/java/ui/filters/FiltersTest.java (hunk 2 above)
Comment (E-Kel) [reply src/test/java/ui/filters/FiltersTest.java:38]: 'cause the CollectionUtils class doesn't load with maven.  I'll uncomment it.

File: src/test/java/ui/filters/FiltersTest.java (hunk 3 above)
Comment (E-Kel) [reply src/test/java/ui/filters/FiltersTest.java:45]: dunno, I didn't delete it. 
I'll try to return it.

File: src/test/java/ui/filters/FiltersTest.java (hunk 2 above)
Comment (4etverochka) [reply src/test/java/ui/filters/FiltersTest.java:38]: It loaded previously.  What happened?

File: src/main/java/ui/page/checkoutStages/AddressPage.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.codeborne.selenide.SelenideElement;
+import org.openqa.selenium.By;
+
+import static com.codeborne.selenide.Selenide.$;
+
+public class AddressPage implements CheckoutButton {
+
+    private SelenideElement proceedToCheckout = $(By.xpath("//button[@name='processAddress']"));
+
```
Comment (DashaLevchenko) [inline src/main/java/ui/page/checkoutStages/AddressPage.java:11]: You can use selenide for xpath, like this:
$(byXpath("//div[text()='Login']"))

File: src/main/java/ui/constants/URL.java [hunk 7]
```diff
@@ -5,6 +5,8 @@ private URL() {
     }
 
     public static final String BASE_URL = "http://automationpractice.com/index.php";
+    public static final String PRODUCT_URL = "http://automationpractice.com/index.php?id_product=2&controller=product";
+
```
Comment (DashaLevchenko) [inline src/main/java/ui/constants/URL.java:9]: Maybe need to rename constant, like PRODUCT_URL_ID_2

File: src/main/java/ui/page/ProductPage.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import com.codeborne.selenide.SelenideElement;
+import org.openqa.selenium.By;
+
+import static com.codeborne.selenide.Selenide.$;
+
+public class ProductPage {
+    private SelenideElement addToCartButton = $(By.xpath("//p[@id='add_to_cart']"));
+    private SelenideElement closePopupIcon = $(By.xpath("//span[@class='cross']"));
+    private SelenideElement cartLable = $(By.cssSelector(".shopping_cart > a"));
+
```
Comment (DashaLevchenko) [inline src/main/java/ui/page/ProductPage.java:12]: Need to rewrite without "By.". "By." it's usage of selenium, not selenide.
Example:
$(".shopping_cart > a")

File: src/main/java/ui/page/checkoutStages/MainPage.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
+package ui.page.checkoutStages;
+
+import com.codeborne.selenide.SelenideElement;
+import org.openqa.selenium.By;
+
+import static com.codeborne.selenide.Selenide.$;
+
+public class MainPage {
+    private SelenideElement productPhoto = $(By.cssSelector("#add_to_cart"));
+
```
Comment (DashaLevchenko) [inline src/main/java/ui/page/checkoutStages/MainPage.java:10]: "#add_to_cart" it's id, not selector.
 In addition,   "By.cssSelector" it's usage of selenium, need to rewrite with usage selenide

File: src/main/java/ui/page/checkoutStages/PaymentPage.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+public class PaymentPage {
+
+    private SelenideElement PayByBankWireButton = $(By.cssSelector(".bankwire"));
+    private SelenideElement PayByCheckButton = $(By.cssSelector(".cheque"));
+    private SelenideElement totalPriceFieldBeforePayment = $(By.xpath("//span[@id='total_price']"));
+    private SelenideElement totalPriceFieldAfterPayment = $(By.cssSelector("#amount"));
+    private SelenideElement confirmOrderButton = $(By.xpath("//span[contains(text(),'I confirm my order')]"));
+    private SelenideElement orderConfirmedMessage = $(By.xpath("//p[@class='alert alert-success']"));
+
```
Comment (DashaLevchenko) [inline src/main/java/ui/page/checkoutStages/PaymentPage.java:17]: Need to rewrite with usage selenide

File: src/main/java/ui/page/checkoutStages/ShippingPage.java [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.SelenideElement;
+import org.openqa.selenium.By;
+
+import static com.codeborne.selenide.Selenide.$;
+
+public class ShippingPage implements CheckoutButton {
+    private SelenideElement checkboxTermsOfService = $(By.cssSelector("#cgv"));
+    private SelenideElement proceedToCheckout = $(By.xpath("//button[@name='processCarrier']"));
+
```
Comment (DashaLevchenko) [inline src/main/java/ui/page/checkoutStages/ShippingPage.java:12]: Need to rewrite with usage selenide

File: src/main/java/ui/page/checkoutStages/SummaryPage.java [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+import com.codeborne.selenide.SelenideElement;
+import org.openqa.selenium.By;
+
+import static com.codeborne.selenide.Selenide.$;
+
+public class SummaryPage implements CheckoutButton {
+    private SelenideElement proceedToCheckout = $(By.cssSelector(".standard-checkout > span"));
+    private SelenideElement totalPriceField = $(By.cssSelector("#total_price"));
+    private SelenideElement signOutButton = $(By.cssSelector(".logout"));
+
```
Comment (DashaLevchenko) [inline src/main/java/ui/page/checkoutStages/SummaryPage.java:12]: Need to rewrite with usage selenide

File: src/main/java/ui/constants/CheckoutConstants.java [hunk 13]
```diff
@@ -1,10 +1,14 @@
 package ui.constants;
 
 public class CheckoutConstants {
-    public static String TOTAL_PRICE = "";
+    private static String TOTAL_PRICE = "";
+    public static final String SUCCESS_ORDER_MESSAGE = "Your order on My Store is complete.";
```
Comment (4etverochka) [inline src/main/java/ui/constants/CheckoutConstants.java:5]: public static final = constant, move please to another class

File: src/main/java/ui/page/checkoutStages/PaymentPage.java [hunk 14]
```diff
@@ -13,29 +15,38 @@
...
-        return (orderConfirmedMessage.isDisplayed());
+        return (orderConfirmedMessage1
+                .shouldBe(Condition.visible)
+                .getText()
+                .equals(SUCCESS_ORDER_MESSAGE)
+                ||
+                orderConfirmedMessage2
+                        .shouldBe(Condition.visible)
+                        .getText()
+                        .equals(SUCCESS_ORDER_MESSAGE));
```
Comment (4etverochka) [inline src/main/java/ui/page/checkoutStages/PaymentPage.java:50]: better to write` || orderConfirmedMessage2` and then space it

File: src/main/java/ui/constants/CheckoutConstants.java (hunk 5 above)
Comment (4etverochka) [reply src/main/java/ui/constants/CheckoutConstants.java:7]: Setter should not be is constant class

File: src/test/java/ui/filters/FiltersTest.java (hunk 2 above)
Comment (E-Kel) [reply src/test/java/ui/filters/FiltersTest.java]: I solve this by resetting JDK in the project environment. Now everything is ok.

**Pull request merged**